test/CurveThreeTokens.test.ts

```diff
@@ -0,0 +1,323 @@
+// --- Imports ---
```

Comment:
_(arbitrary comment location to start a thread)_

> Existing tests will fail with arbitrum set, as the addresses defined in the test files are tightly coupled to mainnet, but the new test file supports both

Since we're now covering two chains, what if we organize things like this instead:
- in `contracts`, create `mainnet`, `arbitrum`, and `shared` folders, and move contracts into their appropriate folders (where `shared` holds the other existing folders)
- same for `scripts`, that way deploy scripts are also organized by chain
- same for `tests`, so test files are also organized by chain
- in `package.json` add a script called `"test:mainnet": TEST_CHAIN_FORK=mainnet yarn test test/mainnet/**/*`, and a similar `test:arbitrum` command
- in `package.json`, update the test script to run `TEST_CHAIN_FORK=mainnet yarn test:mainnet; TEST_CHAIN_FORK=arbitrum yarn arbitrum` 

A nice side effect is that this lets us remove `TEST_CHAIN_FORK` from the env file, and now all tests should always pass. This might be a bit tedious to set up since a bunch of imports will break, but if you use VSCode their auto-update-imports feature should help

Reply:
I see this test file can work for both mainnet and arbitrum, so one downside of this is that this test file either (1) needs to be copy/pasted into both folders, or (2) re-architected so the test suite is generic and once can import the other and change the addresses

I'm ok with either, but if you do the copy/paste approach add some comments to the top of each file explaining that it's copy/pasted and any changes should be made in both files

Reply:
Updated with the above, but for `contracts` and `scripts` I created folders `mainnet`, `shared`, and `generic` (instead of `arbitrum`), as the new contract and script I've added in this PR can be used for both mainnet and arbitrum.

For the tests, I created a generic test file that could be re-used for mainnet and arbitrum

Reply:
Love it, looks great, but looks like the `YearnCrv3Crypto` tests are failing now?

![image](https://user-images.githubusercontent.com/17163988/153948507-7761313b-83e9-4efc-824e-3b99e75b1db6.png)
![image](https://user-images.githubusercontent.com/17163988/153948535-41a93ea7-818d-4cd5-87da-5fc4dce7b7c5.png)

Reply:
oooh! This is because we need to reset the state between test suites, with the reset I used in my other PR currently open https://github.com/ScopeLift/cozy-triggers/pull/11. I'll add it here as well